TODO.md

```diff
@@ -0,0 +1,26 @@
+## TODO
+
+* Add datapoints in provisioning logic for Influx
+* Be more specific in JSON Schema
+* Load things from NextToken when retreiving spot price history
+* Overwrite UserData with temporary taskcluster credentials
+  * Include the API endpoint for the slave to hit the hello
+    endpoint
+  * include the instanceId scope!
+* Encrypt UserData using opengpg.js
+* Send pulse messages for
+  * killing an instance
+  * cancelling a spot request
+  * creating a spot request
```

Comment:
Maybe, just with API end-points your todo says to make I don't see the use-case for anybody but the provisioner logic to mess with spot requests...

In the [original design](https://etherpad.mozilla.org/jonasfj-taskcluster-aws-provisioner-strategies) we discussed following exchanges:
```
worker-type-added    (message is status)
worker-type-updated  (message is status)
worker-type-removed  (message is status)
state-updated (message is #capacity, #pending, #nodes, #running)
```
We obviously can't post the worker definition with `worker-type-*` messages, but we can post the state as best we know it. The idea was that the web nodes could store the state from `state-updated` in memory and return it whenever someone asks for the state by API, then people can either poll for the state of listen on pulse.
(State is naturally published after each provisioning iteration when the state has been updated).